system/lib/libc/musl/arch/emscripten/bits/alltypes.h

```diff
@@ -97,28 +97,31 @@ typedef struct { union { int __i[9]; unsigned __s[9]; } __u; } pthread_attr_t;
 #ifdef __EMSCRIPTEN__
 // For mutex implementation in Emscripten, need to use an extra seventh control field
 // to hold a temporary futex wait & wake location, designated as mutex->_m_addr.
-typedef struct { union { int __i[7]; void *__p[7]; } __u; } pthread_mutex_t;
+typedef struct { union { int __i[8]; volatile int __vi[8]; volatile void *__p[8]; } __u; } pthread_mutex_t;
```

Comment:
Done in https://github.com/kripken/emscripten/pull/4813/commits/d08eb3da9f8f7fb7bea02fba1ac97e455eb63cc3